File: 프로그래머스/lv0/120875. 평행/평행.py
def get_inclination(dotA, dotB):
    dx = dotB[0] - dotA[0]
    dy = dotB[1] - dotA[1]
    return dy / dx

def solution(dots):
    dots.sort(key = lambda x:x[0])
    a, b, c, d = dots

    d1 = get_inclination(a, b)
    d2 = get_inclination(c, d)

    if d1 == d2:
        return 1
    return 0

File: 프로그래머스/lv0/120875. 평행/평행.swift
func isParallel(_ a1: [Int], _ a2: [Int], _ b1: [Int], _ b2:[Int]) -> Bool { 
    return abs(Double(a2[1] - a1[1]) / Double(a2[0] - a1[0])) == abs(Double(b2[1] - b1[1]) / Double(b2[0] - b1[0]))
}

func solution(_ dots:[[Int]]) -> Int {
    let points = dots.sorted { $0[1] < $1[1] }

    return isParallel(points[0], points[1], points[2], points[3]) || isParallel(points[0], points[2], points[1], points[3]) || isParallel(points[0], points[3], points[1], points[2]) ? 1 : 0
}